core/src/test/java/org/infinispan/security/RolePermissionTest.java

```diff
@@ -0,0 +1,102 @@
+package org.infinispan.security;
+
+import java.security.PrivilegedAction;
+import java.security.PrivilegedExceptionAction;
+import java.util.Optional;
+
+import javax.security.auth.Subject;
+
+import org.infinispan.configuration.cache.AuthorizationConfigurationBuilder;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalAuthorizationConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.security.impl.IdentityRoleMapper;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Test;
+
+@Test(groups="functional", testName="security.RolePermissionTest")
+public class RolePermissionTest extends SingleCacheManagerTest {
```

Comment:
this test can be improved. If I understand correctly, this test checks if the subject has the permission and the role.  So, it makes sense to add tests for:
* check correct permission and no role
* check wrong permission and no role => throw exception
* check correct permission and correct role
* check correct permission and wrong role => throw exception
* check wrong permission and correct role => throw exception
* check wrong permission and wrong role => throw exception